### DISCUSSION TITLE
Discussions are for Everyone!

### DISCUSSION THREAD

**Discussion by TimA314** [TimA314/dev-training-ground#16] [Announcements]
Feel free to discuss your questions or concerns here. There is no such thing as a stupid question! Everyone starts somewhere!

**Comment by ritiksaxena124** [comment]
I was thinking there should be a practice section where students can practice their knowledge.
like MCQ's

**Reply by TimA314** [reply]
Great idea, Lets start by creating issues for basic tasks that will lead to this. Would we want to build a backend for this? potentially using node.js?

**Reply by ritiksaxena124** [reply]
Node.js would be great. 
Let's create some issues for the front-end for the Quiz.
we also need to make this website mobile-friendly. so that folks can access the quiz through their mobile devices as well.

**Reply by TimA314** [reply]
I agree.  The website is not very responsive at the moment. Would you be willing to create some issues for these and add a good first issue label. Would be great to get more people involved if they would like. However, for starters would be good if you assign yourselft to tasks you would like to start on.

**Reply by ritiksaxena124** [reply]
I will work on the UI of the Quiz.

**Comment by ritiksaxena124** [comment]
Hey Tim, should we add backend to our application. or should we create our front end for the whole application first?

**Reply by TimA314** [reply]
I do like the idea of a backend. Does Node.js and Express.js seem like a good choice?  I am hosting the front end using firebase and it would be easy to host a node server there as well.  I could make  basic example CRUD calls and it would make it easy for those to be custimized. Another question is should this be in the same repo or a new repo?

**Reply by ritiksaxena124** [reply]
Node.js and Express.js would be great choices. I think we should keep both front-end and back-end in different repos. it will make it easy for the new contributors to contribute. as if a contributor wants to contribute to the front-end can easily be able to contribute to it and the same for the backend.

**Reply by TimA314** [reply]
What do you think about the database? would SQL express be a good option? Open to ideas. Here is a node/express repo I started. https://github.com/TimA314/dev-training-ground-server

**Comment by TimA314** [comment]
maybe for now lets hard code data in a separate data file in the server app and will discuss this more later.

**Reply by ritiksaxena124** [reply]
yup, that's sounds great!!! Let's do this.
